### PR TITLE
Add new functions, revise existing ones

### PR DESCRIPTION
**Pull request by dpikhulya**
Introduces functions that can simplify writing tests written as [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) (these functions return `Promise` instances for them to be usable with the `await` keyword):
- `asyncDelay` — performs an asynchronous delay (can be used like `await asycDelay(100);`
- `asyncCondition` — makes an asynchronous delay until a condition is met. This is a `Promise`-based analog of an existing `waitForCondition` function, which was made deprecated in this PR.

The `checkNodeList` function was renamed to `checkNodes` to make usages more compact and avoid a dubious "list" term.

Besides, contains the following changes in the `Checkers` hash:
- The `childNodesByFilters` option of `Checkers.element` was simplified to accept only one pair of `{filter, checkers}` object, and was renamed to `childNodesByFilter` to reflect this. The multi-filter scenario is likely quite rare (if needed at all), and can be simulated with the `checkers` parameter of `Checkers.element`, if needed.
- Added `shadowNodes` and `shadowNodesbyFilter` options for `Checkers.element` function to check elements inside of element's shadow DOM.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "polymer",
     "test"
   ],
-  "version": "0.4.1",
+  "version": "0.6.0",
```
Comment (dpikhulya) [inline package.json:10]: Version 0.5.0 was [released](https://github.com/SpineElements/spine-test-helpers/releases/tag/v0.5.0) without updating this file.

File: test/spine-test-helpers_test.html [hunk 2]
```diff
@@ -58,63 +58,202 @@
...
+     *                               instance creation at the time when this method is invoked
+     * @param {String=} message      an optional message to include into the assertion message
+     */
+    assertElapsedTimeCorrect(timeoutDelay, message) {
+      if (this._endTime) {
+        throw new Error('This method can be invoked only once per instance');
+      }
+      this._endTime = Date.now();
+      const messagePrefix = message ? `${message}: ` : '';
+      const maxElapsedTime = timeoutDelay * 1.05 + 50;
```
Comment (armiol) [inline test/spine-test-helpers_test.html]: Let's have a comment explaining.

File: test/spine-test-helpers_test.html [hunk 3]
```diff
@@ -58,63 +58,202 @@
...
-          ['non-empty-text-nodes', 'displayed-nodes', 'nodes-with-visibility-hidden'],
-          ['nodes-with-display-none', 'comment-nodes', 'whitespace-nodes']
-      );
+    suite('Async tests helper functions', () => {
+      test('asyncDelay() delays execution by a requested amount', async () => {
+        const checkWithDelay = async delay => {
+          const timeBefore = Date.now();
+          await asyncDelay(delay);
+          const timeAfter = Date.now();
+          expect(timeAfter - timeBefore).to.be.within(delay, delay * 1.1 + 50);
```
Comment (armiol) [inline test/spine-test-helpers_test.html]: Same as above.

**Review by armiol — APPROVED** [review]
@dpikhulya LGTM with the comments to address.

**Pull request merged**